### PR TITLE
update support from mailto to url

### PR DESCRIPTION
**Pull request by estherjsuh**
## Change description

Update support for AVL issues to show URL (`tahoe-domain/help`)  instead of email `technical+{{ COMMON_DEPLOYMENT }}@appsembler.com` when `support_email` is not set.

Currently, our customer success team is providing T1 support for AVL issues from learners when the issues should be directed to the customer.   

Issue: https://appsembler.atlassian.net/browse/RED-2278

Linked to PRs in:

- ~~`configuration` - [PR-375](https://github.com/appsembler/configuration/pull/375)~~
-  ~~`edx-configs` - [PR-1144](https://github.com/appsembler/edx-configs/pull/1144)~~

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [RED-2278](https://appsembler.atlassian.net/browse/RED-2278) 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by melvinsoft — DISMISSED** [review]
@estherjsuh Thanks for implementing the change here. I'm not familiar with the XBlock, but the changes makes total sense to me.

Please hold off merging until getting approval from @tkeemon | @OmarIthawi | @ahmedaljazzar since the know the XBlock more.

My only concern is about the front end, what happens when enterprise customers actually want to use an email, the link will probably break, but maybe that's ok and we are encouraging the use an URL rather than an email.

**Review by thraxil — DISMISSED** [review]

File: launchcontainer/launchcontainer.py
Comment (OmarIthawi) [inline launchcontainer/launchcontainer.py]: Step 1) Remove the default support email, so it works only if the customer sets it explicitly.

```py
    support_email = String(
        display_name='Tech support email',
        default=None,
        scope=Scope.content,
        help=("Email address of tech support for AVL labs."),
    )
```

File: launchcontainer/static/js/src/launchcontainer.js
Comment (OmarIthawi) [inline launchcontainer/static/js/src/launchcontainer.js]: Step 2) We still need to show the support email to avoid breaking existing content, but only show it when it's been set. Now it defaults to `None` it will be mostly not set.


```js
if  (supportEmail) {
  $launch_notification.append('Try contacting <a href="mailto:' + supportEmail + '">' + supportEmail + '</a>')
} else {
  $launch_notification.append('<a href="/help">Contact us for support.</a>')
}

File: launchcontainer/launchcontainer.py
Comment (OmarIthawi) [inline launchcontainer/launchcontainer.py:302]: Step 3) We can hardcode support URL to `/help` because it's a standard Open edX URL.

File: launchcontainer/static/js/src/launchcontainer.js
Comment (OmarIthawi) [inline launchcontainer/static/js/src/launchcontainer.js]: Step 4) Same here as done in Step 2

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Thanks @estherjsuh for making a such detailed PR! This is an XBlock that's been used in production. So we have little room for change.

I've suggested a less impactful change that maintains the compatibility with existing content.

The change is effectively a redo of the PR, so please let me know if you have concerns of feedback that you think we shouldn't go with the suggestions.

**Comment by estherjsuh** [comment]
@OmarIthawi @melvinsoft 
Per Omar's suggestion, I reverted `support_url` updates back to `support_email` (with the default value for `support_email` as None).
We no longer have a `support_url` as an instance. Instead, the `/help` endpoint will be directed from the frontend when `support_email` is not set (preserving this field for enterprise customers).


@melvinsoft Added TJ, Ahmed and also Bryan as reviewers 🥳 

Please review the recent changes. Thanks!

File: launchcontainer/launchcontainer.py
Comment (iamjazzar) [inline launchcontainer/launchcontainer.py]: `data['support_email']` will cause a `KeyError` if `support_email` is not in the dict. You can use
```suggestion
        self.support_email = data.get('support_email', '').strip() or None
```

**Review by iamjazzar — CHANGES_REQUESTED** [review]
I have one comment on the new python changes, I'll leave the frontend side for another developer to check.

File: launchcontainer/launchcontainer.py
Comment (estherjsuh) [reply launchcontainer/launchcontainer.py]: @ahmedaljazzar 
I was avoiding `AttributeError: 'NoneType' object has no attribute 'strip'` here because we are setting the default value for `support_email` to None.

However, I actually think I can leave this as the original code (`self.support_email = data['support_email'].strip()`) because we are converting `None` values to `''` [here](https://github.com/appsembler/xblock-launchcontainer/blob/master/launchcontainer/launchcontainer.py#L260-L264).

Please let me know what you think.

**Comment by estherjsuh** [comment]
@ahmedaljazzar thank you for the review! I replied to your comment

File: launchcontainer/launchcontainer.py
Comment (iamjazzar) [reply launchcontainer/launchcontainer.py]: @estherjsuh You took a nice step forward here, but technically we still have the same issue with the new change. As python tries to evaluate the condition on empty data (`data={}`):
```python
 not data['support_email']
```
will throw that `AttributeError` again.

The change that I suggested achieves the end goal outlined; Python will do the following:
- Try to get `support_email` from data
- If no key found it will return `''`
- Strip the fetched value. This will result to either `email@appsembler.com` or `''` in case fetched `support_email` is `None` or empty.
- If the string we received is empty, we set `self.support_email` to None (That's what `or` does here). 

So no risk of `AttributeError` again.

> NOTE: reverting to `self.support_email = data['support_email'].strip()` is not going to solve the issue


Here's a little snippet, try to uncomment these lines to see it work in action

```python

data = {
    # 'support_email': ' email@appsembler.com    '
}

class A(object):
    def __init__(self):
        # self.support_email = None if not data['support_email'] else data['support_email'].strip()
        # self.support_email = data['support_email'].strip()
        self.support_email = data.get('support_email', '').strip() or None


if __name__ == '__main__':
    a = A()
    print(a.support_email)
```

Comment (OmarIthawi) [reply launchcontainer/launchcontainer.py]: Applying Ahmed's suggestion:

```suggestion
            self.support_email = data.get('support_email', '').strip() or None
```

File: launchcontainer/static/js/src/launchcontainer.js
Comment (OmarIthawi) [inline launchcontainer/static/js/src/launchcontainer.js]: A little bit of rewording and open in a new page:

```suggestion
            $final_msg += 'at <a href="/help" rel="noreferrer" target="_blank">help section</a>';
```

File: launchcontainer/static/js/src/launchcontainer.js
Comment (OmarIthawi) [inline launchcontainer/static/js/src/launchcontainer.js]: Please refactor the whole if statement to become:

```js
timeoutMessage = setTimeout(function () {
    var $support_link = '<a href="/help" rel="noreferrer" target="_blank">Contact us for support.</a>';
    if (supportEmail) {
        var $support_link = 'Try contacting <a href="mailto:' + supportEmail + '">' + supportEmail + '</a>';
    }

    $launch_notification.append(
        '<br /><br />Lab taking a long time to load? Try contacting ' + $support_link
    );
}, 120 * 1000); // show message after 2min
```

**Review by OmarIthawi — DISMISSED** [review]
Thanks @estherjsuh! Looks great now! I've added minor suggestions. Please feel free to address/leave them as-is.

**Review by iamjazzar — DISMISSED** [review]

**Review by iamjazzar — DISMISSED** [review]
Approving Python changes.

File: launchcontainer/static/js/src/launchcontainer.js
Comment (OmarIthawi) [inline launchcontainer/static/js/src/launchcontainer.js:56]:
```suggestion
        clearTimeout(timeoutMessage);
        timeoutMessage = setTimeout(function () { 
```

**Review by OmarIthawi — DISMISSED** [review]
Thanks @estherjsuh! one more change!

**Review by OmarIthawi — APPROVED** [review]

**Review by bryanlandia — APPROVED** [review]
Looks good to me!

**Pull request merged**
